enhancements/arbiter-clusters.md

```diff
@@ -0,0 +1,670 @@
+like the Master+Arbiter in this enhancement.
+
+This is a nice to have before we go GA and we should be looking into adding it
+to OLM.
+
+### Workflow Description
+
+#### For Baremetal IPI Installs
+
+1. The user creates an `install-config.yaml` like normal.
+2. The user defines the `installConfig.controlPlane` field with `2` replicas.
+3. The user then enters the new field `installConfig.arbiter` defines the
+   arbiter node pool and it's replicas set to `1` and the machine type desired
+   for the platform chosen.
+4. The user then enters the empty information for
+   `installConfig.controlPlane.platform.baremetal` and
+   `installConfig.arbiter.platform.baremetal`.
+5. The user adds the appropriate hosts to the `platform.baremetal.hosts` array,
+   2 master machines with the `role` of `master` and 1 arbiter machine with the
+   `role` of `arbiter`, the typical flow for bare metal installs should be
+   followed.
+6. The user runs `openshift-install create manifests`
+7. The installer creates a new `arbiter-machine-0` Machine same as
+   `master-machine-{0-1}` is currently generated.
+8. The arbiter machine is labeled with
+   `machine.openshift.io/cluster-api-machine-role: arbiter` and
+   `machine.openshift.io/cluster-api-machine-type: arbiter` and set with the
+   taint for `node-role.kubernetes.io/arbiter=NoSchedule`.
+9. Arbiter is treated like a master node, in that the installer creates the
+   `arbiter-ssh.yaml` and the `arbiter-user-data-secret.yaml`, mirroring a
+   normal master creation.
+10. The installer sets the `ControlPlaneTopology` to `HighlyAvailableArbiter`
+    and `InfrastructureTopology` to `HighlyAvailable`
+11. The user then begins the install via `openshift-install create cluster`
+
+#### During Install
+
+1. The CEO will watch for new masters and the arbiter role
+2. CEO will create the operand for the etcd deployments that have tolerations
+   for the arbiter
+3. Operators that have tolerations for the arbiter should be scheduled on the
+   node
+4. The install should proceed as normal
+
+### API Extensions
+
+#### Installer API Change
+
+The `installConfig` will include a `installConfig.arbiter` object for a
+`MachinePool` to configure arbiter infra structure.
+
+#### OCP Config API Change
+
+The Infrastructure config fields for `ControlPlaneTopology` will support the new
+value of `HighlyAvailableArbiter`
+
+`InfrastructureTopology` will be left as `HighlyAvailable` in this situation
+since the existing rule is we only highlight when we are dealing with
+`SingleReplica` workers. In this instance, at minimum 2 masters will be able to
+schedule worker workloads, thus follow the same pattern as a 3 Node cluster.
+This makes the most sense for now since having `InfrastructureTopology` denote a
+halfway point between single and three might be a moot distinction for workers.
+However, we might need to revisit this as a product team might assume a
+`InfrastructureTopology: HighlyAvailable` == `3 Workers`.
```

Comment:
This shouldn't be a problem as a cluster with 3 control plane nodes and 2 workers would still have `InfrastructureTopology: HighlyAvailable`.

Out of curiosity, is it implied and/or required that clusters with an arbiter node are also created without workers (i.e. mastersSchedulable)?

Reply:
OK, so we don't need to have a validation that mastersSchedulable is enabled, and the default for it should continue to be based only on whether the number of workers is <2 :+1: